### PR TITLE
Jenkinsfile: build the docs in docker instead

### PR DESCRIPTION
**Pull request by tobsan**
Removes the requirement to keep up with dependencies on the build
servers, which is nice, plus the Dockerfile acts as documentation for
what's needed to build the docs.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

**Comment by tobsan** [comment]
Thanks @rpannek for the Dockerfile!

File: Jenkinsfile [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 pipeline {
     agent {
-        node { label 'Sphinx' }
+        dockerfile { label 'DockerCI' }
```
Comment (demonicblue) [inline Jenkinsfile]: Looks good but you can specify the label for nodes supporting Docker in the global jenkins settings instead.  _Manage Jenkins_ -> _Configure System_ -> _Docker Label_

**Review by demonicblue — COMMENTED** [review]
Looks good. Just a small comment.

File: Jenkinsfile (hunk 1 above)
Comment (tobsan) [reply Jenkinsfile]: Updated.

**Review by demonicblue — APPROVED** [review]

**Pull request merged**